view/frontend/web/js/view/payment/method-renderer/card-method.js

```diff
@@ -51,6 +51,7 @@ define([
 
             confirmIntern: function () {
                 $('body').trigger('processStart');
+                this.createIntent();
```

Comment:
Why do we call `createIntent` on `confirmIntent`? 

also is the `confirmIntern` method correctly named?

Reply:
This was refractored to create intent just before order placement, to solve the issue with wrong discount code calculated during the checkout. This method is called on order placement. How should this method be renamed?